sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/SequenceSizeError.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "400"}
+spark-rapids-shim-json-lines ***/
+package org.apache.spark.sql.rapids.shims
+
+import org.apache.spark.sql.errors.QueryExecutionErrors
+
+object SequenceSizeError {
+  def getTooLongSequenceErrorString(sequenceSize: Int, functionName: String): String = {
+    QueryExecutionErrors.createArrayWithElementsExceedLimitError(functionName, sequenceSize)
```

Comment:
We should move this to RapidsErrorUtils 

The way I would do it is remove the `TOO_LONG_SEQUENCE` from the `GetSequenceSize.scala` altogether 

Then introduce another trait `RapidsErrorUtilsForSequence` with a method `tooLongSequenceError`. The versions of 320 will have it's own implementation returning the hardcoded message "Too long..."  334-352 will have it's own implementation returning "Unsuccessful try..." and 400 will have it's own where it returns the QueryExecution.createArrayWithElementsExceedLimitErrors

Reply:
I think we're talking at cross purposes.  Or maybe I'm simply unable to grok the suggestion yet.

>  The versions of 320 will have it's own implementation returning the hardcoded message "Too long..." 334-352 will have it's own implementation

This largely describes what my patch currently does, save for moving it into `RapidsErrorUtils`.

The reason (I think) I can't move this into `RapidsErrorUtils` is that the error messages differ within the same major Spark version.  

The error messages are split as follows:
1. `Too long sequence found`:
    1. 3.2.*
    1. 3.3.[x<4]
    1. 3.4.[x<2]
    1. 3.5.0
2. `Unsuccessful try to create array`:
    1. 3.3.4
    1. 3.4.[2-3]
    1. 3.5.[1-2]
3. `Can't create array with...`:
    1. Only 4.0.  

The `RapidsErrorUtils` shims are grouped by version, as follows:
1. All 3.2.* together.
2. All 3.3.* except 3.3.*db
3. All 3.3.[0,2]db
4. 3.4.[0-3] + 3.5.[0-2] + 4.0.0.

Now if I'm trying to accommodate the correct error message for, say, `Spark 3.4.2`, I can't because there's only one `RapidsErrorUtils` for all of Spark 3.4.x (and that class happens to affect all 3.5.x as well as 4.0).  

Is the suggestion to further slice up `3.4.*`'s `RapidsErrorUtils`?  We will then have to also slice up the same for `3.3.x` and `3.5.x` as well, with code duplicated everywhere.  This doesn't sound productive to me.  

Maybe I've missed something.  Perhaps we should discuss this offline, and update the result on this bug.

Reply:
This refactor is turning into a rats' nest.  When the next shim needs to be added, and things need to be split further, I think it's going to be unreadable.

`RapidsErrorUtilsBase`, used in `33xdb` is a very misleading name for the errors shim.  It seems to apply only in `33x`, while its name suggests that it's the base-class for all the `RapidsErrorUtils`.  This is painful.

I'm going to try to add this with as little collateral damage as I can.

Reply:
I have fixed this.  As discussed, I have refactored the `RapidsErrorUtils` into more concrete versions, and used mix-ins for the new behaviours.